src/Database/Schema/MysqlSchema.php

```diff
@@ -92,7 +92,7 @@ protected function _convertColumn($column)
             $precision = (int)$precision;
         }
 
-        if (in_array($col, ['date', 'time', 'datetime', 'timestamp'])) {
+        if (in_array($col, ['date', 'time', 'datetime', 'timestamp', 'year'])) {
             return ['type' => $col, 'length' => null];
```

Comment:
If a table has a `YEAR` column and we reflect it into a `TableSchema` object, can we then generate the correct SQL again?